src/prefect/tasks/shell.py

```diff
@@ -109,4 +109,4 @@ def run(self, command: str = None, env: dict = None) -> str:
         if self.return_all:
             return lines
         else:
-            return line
+            return lines[-1]
```

Comment:
Hey @RoyalTS ! Good catch -> I think the appropriate fix here is to pre-define the `line` variable in line 95, prior to the iteration.  The reason this is failing tests is because the `lines` list is _only_ appended to whenever `return_all` is `True`.